pkg/provisioner/ironic/ironic.go

```diff
@@ -239,7 +239,7 @@ func (p *ironicProvisioner) listAllPorts(address string) ([]ports.Port, error) {
 		opts.Address = address
 	}
 
-	pager := ports.List(p.client, opts)
+	pager := ports.ListDetail(p.client, opts)
```

Comment:
If we only need a couple of fields, we can request them from `List()` by specifying it in the [`Fields` field](https://github.com/gophercloud/gophercloud/blob/master/openstack/baremetal/v1/ports/requests.go#L34-L35) of `opts` on line 236. With a big list that would be more efficient.